pkg/system/version.go

```diff
@@ -1,4 +1,61 @@
 package system
 
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"golang.org/x/mod/semver"
+)
+
 var Version string
 var CommitSHA string
+
+type VersionConfig struct {
+	version        string
+	minimumVersion string
+}
+
+func NewVersionConfig(minimumVersion string) (*VersionConfig, error) {
+	if Version != "" && !semver.IsValid(Version) {
+		return nil, fmt.Errorf("invalid embedded version: %s", Version)
+	}
+
+	if minimumVersion != "" && !semver.IsValid(minimumVersion) {
+		return nil, fmt.Errorf("invalid minimum version: %s", minimumVersion)
+	}
+
+	if Version != "" &&
+		minimumVersion != "" &&
+		semver.Compare(minimumVersion, Version) > 0 {
+		return nil, fmt.Errorf("minimum version %s is greater than embedded version %s", minimumVersion, Version)
```

Comment:
Let's add log statements inside these blocks at least initially to help us debug/troubleshoot